browser/brave_wallet/keyring_service_unittest.cc

```diff
@@ -1164,32 +1168,37 @@ TEST_F(KeyringServiceUnitTest, UnlockResumesDefaultKeyring) {
 }
 
 TEST_F(KeyringServiceUnitTest, GetMnemonicForDefaultKeyring) {
+  // Needed to skip unnecessary migration in CreateEncryptorForKeyring.
+  GetPrefs()->SetBoolean(kBraveWalletKeyringEncryptionKeysMigrated, true);
   KeyringService service(json_rpc_service(), GetPrefs());
-  ASSERT_TRUE(
-      service.CreateEncryptorForKeyring("brave", mojom::kDefaultKeyringId));
+  ASSERT_TRUE(service.CreateEncryptorForKeyring(kPasswordBrave,
+                                                mojom::kDefaultKeyringId));
 
   // no pref exists yet
-  EXPECT_TRUE(GetMnemonicForDefaultKeyring(&service).empty());
+  EXPECT_TRUE(GetMnemonicForDefaultKeyring(kPasswordBrave, &service).empty());
 
   ASSERT_TRUE(service.CreateKeyringInternal(mojom::kDefaultKeyringId,
                                             kMnemonic1, false));
-  EXPECT_EQ(GetMnemonicForDefaultKeyring(&service), kMnemonic1);
+  EXPECT_EQ(GetMnemonicForDefaultKeyring(kPasswordBrave, &service), kMnemonic1);
 
   // Lock service
   service.Lock();
   ASSERT_TRUE(service.IsLocked(mojom::kDefaultKeyringId));
-  EXPECT_TRUE(GetMnemonicForDefaultKeyring(&service).empty());
+  EXPECT_TRUE(GetMnemonicForDefaultKeyring(kPasswordBrave, &service).empty());
 
   // unlock with wrong password
-  ASSERT_FALSE(Unlock(&service, "brave123"));
+  ASSERT_FALSE(Unlock(&service, kPasswordBrave123));
```

Comment:
Nice to see it locks after 1 failed attempt here since that's simpler for state management at this layer.

Reply:
It's not because of 1 failed attempt, it's just still locked because this line try to unlock with wrong password.